plugins/org.yakindu.sct.model.stext/src/org/yakindu/sct/model/stext/validation/STextValidator.xtend

```diff
@@ -307,6 +309,23 @@ class STextValidator extends AbstractSTextValidator implements STextValidationMe
 			}
 		}
 	}
+	
+	@Check(CheckType.FAST)
+	def void checkNotRaisedOutEvent(EventDefinition event) {
+		if (event.direction != Direction.OUT) {
+			return;
+		}
+		var Collection<Setting> usages = EcoreUtil.UsageCrossReferencer.find(event, event.eResource().getResourceSet());
+		var boolean isRaised = usages.findFirst[setting | setting.getEObject().eContainer instanceof EventRaisingExpression] !== null;
```

Comment:
Nice usage of xtend. Even better suited for the task is [`exists`](http://download.eclipse.org/modeling/tmf/xtext/javadoc/2.4/org/eclipse/xtext/xbase/lib/IterableExtensions.html#exists(java.lang.Iterable,%20org.eclipse.xtext.xbase.lib.Functions.Function1)).